week6/week6_prepartion_database/Program.cs

```diff
@@ -0,0 +1,271 @@
+  }
+  // it only updates the product available but it doesnot set the id to that id that already not exists.. ???
+  await using var connection = new MySqlConnection(configuration.GetConnectionString("Default"));
+  var productUpdated = await connection.ExecuteAsync($"UPDATE dapper.products SET name=@name, price=@price,description=@description WHERE id={id}", product);
+  // var productId = await connection.ExecuteAsync($"UPDATE dapper.products SET name={product.Name}, price={product.Price} WHERE id={product.Id}", product);// ##### seems logical, but why this does not work?? 
+  if (productUpdated == 0)
+  {
+    return Results.NotFound($"this id: {id} not found");//404 not found
+  }
+  //##### id is set to 0 ?? doesnot work??
+  return Results.Accepted($"/product/{id}", product); // 202 Accepted
+});
+
+
+/*
+<questions>
+  1.Implement an endpoint /product/{id} that takes that id and tries to delete the product with that ID
+  2.Think about what should the endpoint return.
+  3.What happens if there is no such product? What is the return of that endpoint?
+</qustions>
+*/
+app.MapDelete("/products/{id}", async (IConfiguration configuration, int id) =>
+{
+  if (id < 0)
+  {
+    return Results.BadRequest("id does not exists");//400 badrequest
+  }
+  await using var connection = new MySqlConnection(configuration.GetConnectionString("Default"));
+  var deletedProduct = await connection.ExecuteAsync("DELETE FROM dapper.products WHERE id=@id", new { ID = id });//#####
+  if (deletedProduct == 0)
+  {
+    return Results.NotFound($" this product id:{id} not found");//404 not found
+  }
+  return Results.Ok($"{id} deleted");//200 ok
+});
+
+
+// join table ?? #####
+
+// SELECT *
+// FROM ARTIST AS ART
+// INNER JOIN ALBUM AS ALB
+// ON ART.ARTIST_ID = ALB.ARTIST_ID;
+app.MapGet("/productsjoin", async (IConfiguration configuration) =>
+{
+  using var connection = new MySqlConnection(configuration.GetConnectionString("Default"));
+  // var products = await connection.QueryAsync<Product>("SELECT dapper.products.id, name, price, description, dapper.sales.sales_date  FROM dapper.products join dapper.sales on dapper.products.id= dapper.sales.products_id ");
+  // List<T> genericList = new List<T>();
+  var products = await connection.QueryAsync<Product>("SELECT id, name, price, description FROM dapper.products");
+  var sales = await connection.QueryAsync<Sales>("SELECT *  FROM dapper.sales ");
+  // var salesman = await connection.QueryAsync<Salesman>("SELECT dapper.products.id, name, price, description, dapper.sales.sales_date  FROM dapper.products join dapper.sales on dapper.products.id= dapper.sales.products_id ");
+  return Results.Ok(sales);
+});
+
+
+app.Run();
+
+// public record User(string user, string host)
+// { public User() : this("", "") { } }
+public class User
+{
+  public string? user { get; set; } = "suman";
+  public string? host { get; set; } = "ghimire";
+  public User() { }// shows error without this empty constructor ??
```

Comment:
Do you write `new User()` somewhere? Where is the error?

Reply:
at the end point at lie 34 ( app.MapGet("/users" ), it works only with either user or host at line 48.  passing both user and host at a time then it nees for the empty constructure !